### ISSUE TITLE
Fix dynawo

### ISSUE THREAD

**Issue by cossartque** [dynawo/dynawo#2619]

**Comment by cossartque** [comment]
The compilation works but the results are not as expected.
@GerardThevenon can you give the test case to @rosiereflo ?
Thanks

**Comment by GerardThevenon** [comment]
Build failed on [2619_dynamic_indexing](https://github.com/dynawo/dynawo/tree/2619_dynamic_indexing)

(IECWT4ACurrentSource)

**Comment by rosiereflo** [comment]
> Build failed on [2619_dynamic_indexing](https://github.com/dynawo/dynawo/tree/2619_dynamic_indexing)
> 
> (IECWT4ACurrentSource which uses Multiswitch)

Yup, a bug that I introduced at the last moment. Let me fix that

**Comment by rosiereflo** [comment]
should be ok now

**Comment by GerardThevenon** [comment]
> should be ok now

Unfortunately it looks as if the index doesn't take into account the operation "+1" defined in MultiSwitch.mo
Also the behavior of a step is questionable. Here are some results :

![MicrosoftTeams-image (4)](https://user-images.githubusercontent.com/98324724/206478327-9dbc63e2-fceb-4b3c-ae71-9b6caea23612.png)
![MicrosoftTeams-image (3)](https://user-images.githubusercontent.com/98324724/206478338-dece980d-ba3a-4367-8c38-219fae1ad1a4.png)
![MicrosoftTeams-image (2)](https://user-images.githubusercontent.com/98324724/206478341-71a8a755-9f3c-4909-bc8f-fed83df67db0.png)
![MicrosoftTeams-image (1)](https://user-images.githubusercontent.com/98324724/206478343-b4e78ae7-3689-426e-b6b8-f3846b8cbfe3.png)
![MicrosoftTeams-image](https://user-images.githubusercontent.com/98324724/206478346-21a3695e-58e5-4c4f-9d60-b241b9e3dfc8.png)

**Comment by rosiereflo** [comment]
Ready for a new test!

**Comment by GerardThevenon** [comment]
I got the same result.

**Comment by rosiereflo** [comment]
Are you sure the models were recompiled ?

**Comment by GerardThevenon** [comment]
I didn't want to build-all so I compiled dynawo-build-models IECMultiSwitch

**Comment by GerardThevenon** [comment]
In the end, it works as intended after clean-build-all !!
